lib/tests/config_value_factory_test.cc

```diff
@@ -0,0 +1,63 @@
+#include <catch.hpp>
+#include "test_utils.hpp"
+
+#include <hocon/config_value_factory.hpp>
+#include <internal/values/simple_config_object.hpp>
+
+using namespace hocon;
+using namespace std;
+
+TEST_CASE("convert various types to config values", "[config_value_factory]") {
+    SECTION("true should convert to config_boolean") {
+        auto value = config_value_factory::from_any_ref(true, "");
+        REQUIRE(dynamic_pointer_cast<const config_boolean>(value));
+        REQUIRE(boost::get<bool>(value->unwrapped()) == true);
+    }
+
+    SECTION("false should convert to config_boolean") {
+        auto value = config_value_factory::from_any_ref(false, "");
+        REQUIRE(dynamic_pointer_cast<const config_boolean>(value));
+        REQUIRE(boost::get<bool>(value->unwrapped()) == false);
+    }
+
+    // TODO: Where do config_nulls usually come from? What type might conceivably be converted to them in C++?
+    // Because passing a boost::blank causes one to be produced, but in theory we might want other values to do so,
+    // but because of value semantics, nullptr is out.
```

Comment:
The from_any_ref method is not used internally anywhere, and in all the tests, the only value passed to test config_null is 'null', with no indication of how that 'null' was produced. I feel like we will have to see what our use cases for this method are before we can decide what C++ values should produce config_nulls.

In every case that I have come across, failure to resolve results in an exception being thrown.